gax/src/main/java/com/google/api/gax/rpc/ApiCallContext.java

```diff
@@ -153,6 +156,54 @@
   @BetaApi("The surface for tracing is not stable yet and may change in the future")
   ApiCallContext withTracer(@Nonnull ApiTracer tracer);
 
+  /**
+   * Returns a new ApiCallContext with the given {@link RetrySettings} set.
+   *
+   * <p>This sets the {@link RetrySettings} to use for the RPC. These settings will work in
+   * combination with either the default retryable codes for the RPC, or the retryable codes
+   * supplied through {@link #withRetryableCodes(Set)}. Calling {@link
+   * #withRetrySettings(RetrySettings)} on an RPC that does not include {@link
+   * Code#DEADLINE_EXCEEDED} as one of its retryable codes (or without calling {@link
+   * #withRetryableCodes(Set)} with a set that includes at least {@link Code#DEADLINE_EXCEEDED})
+   * will effectively only set a simple timeout that is equal to {@link
+   * RetrySettings#getInitialRpcTimeout()}. It is recommended to use {@link #withTimeout(Duration)}
+   * if that is the intended behavior.
+   *
+   * <p>Example usage:
+   *
+   * <pre>{@code
+   * ApiCallContext context = GrpcCallContext.createDefault()
+   *   .withRetrySettings(RetrySettings.newBuilder()
+   *     .setInitialRetryDelay(Duration.ofMillis(10L))
+   *     .setInitialRpcTimeout(Duration.ofMillis(100L))
+   *     .setMaxAttempts(10)
+   *     .setMaxRetryDelay(Duration.ofSeconds(10L))
+   *     .setMaxRpcTimeout(Duration.ofSeconds(30L))
+   *     .setRetryDelayMultiplier(1.4)
+   *     .setRpcTimeoutMultiplier(1.5)
+   *     .setTotalTimeout(Duration.ofMinutes(10L))
+   *     .build())
+   *   .withRetryableCodes(Sets.newSet(
+   *     StatusCode.Code.UNAVAILABLE,
+   *     StatusCode.Code.DEADLINE_EXCEEDED));
+   * }</pre>
+   */
+  ApiCallContext withRetrySettings(RetrySettings retrySettings);
```

Comment:
Can we mark these new methods as `BetaApi`. This an addition to a fundamental interface, whatever we add here we will have to stick with for a really long time.

Reply:
Done